src/randoop/JunitFileWriter.java

```diff
@@ -195,58 +183,86 @@ public File writeDriverFile(String driverClassName) {
+      if (testIterator.hasNext()) {
+        out.println(testIterator.next() + ".class");
+        while (testIterator.hasNext()) {
+          out.print(", ");
+          out.println(testIterator.next() + ".class");
+        }
       }
-      if (GenInputsAbstract.init_routine != null)
-        out.println ("    " + GenInputsAbstract.init_routine + "();");
-
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("    TestRunner runner = new TestRunner();");
-      out.println("    TestResult result = runner.doRun(suite(), false);");
-      out.println("    if (! result.wasSuccessful()) {");
-      out.println("      System.exit(1);");
-      out.println("    }");
-      } else {
-      out.println("    new " + driverClassName + "(\"" + driverClassName + "\").runTest();");
+      
+      out.println("})");
+      out.println("public class " + suiteClassName + "{ }");
+    } finally {
+      if (out != null) {
+        out.close();
       }
-      out.println("  }");
-      out.println("");
-      out.println("  public " + driverClassName + "(String name) {");
-      out.println("    super(name);");
-      out.println("  }");
-      out.println("");
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("  public static Test suite() {");
-      out.println("    TestSuite result = new TestSuite();");
-      for (String junitTestsClassName : junitTestSuiteNames) {
-        out.println("    result.addTest(new TestSuite(" + junitTestsClassName + ".class));");
+    }
+    return file;
+  }
+  
+  /*
+   * write non-reflective driver for tests
+   * Note that this relies on method names being generated as "test"+i in writeSubSuite
+   */
+  public File writeDriverFile(List<String> testClassNames) {
+    File dir = this.getDir();
+    File file = new File(dir, masterTestClassName + "Driver.java");
+    PrintStream out = createTextOutputStream(file);
+    try {
+      outputPackageName(out, packageName);
+
+      out.println();
+      out.println("public class " + masterTestClassName + "Driver {");
+      out.println();
+      out.print("  public static void main(String[] args) {");
+      
+      if (GenInputsAbstract.init_routine != null) {
+        out.println ("    " + GenInputsAbstract.init_routine + "();");
       }
-      out.println("    return result;");
-      out.println("  }");
-      } else {
-      out.println("  @Override");
-      out.println("  public void runTest() throws Throwable {");
-      for (String junitTestsClassName : junitTestSuiteNames) {
-        out.println("    new " + junitTestsClassName + "().runTest();");
+      
+      int classIndex = 0; //counter for test class object instances
+      for (String testClass : testClassNames) {
+        String testVariable = "t"+classIndex;
+        out.println(testClass + " " + testVariable + "= new " + testClass + "()");
+        
+        if (methodCountMap.containsKey(testClass)) {
```

Comment:
Under what circumstances might this fail?

If it cannot return false, then please don't put it in an if test that would silently mask an error.
If it can, write a brief comment saying how or why it would return false.